app/javascript/utils/fetch-json.ts

```diff
@@ -1,35 +1,46 @@
 import { camelizeKeys } from 'humps'
 
-export const fetchJSON = <T extends any>(
+export class ApiError extends Error {
+  type: string
+
+  constructor(type: string, message: string) {
+    super(message)
+    this.name = this.constructor.name
+    this.type = type
+    Object.setPrototypeOf(this, ApiError.prototype)
```

Comment:
Why is this needed?

Reply:
For Typescript. It is needed to correctly set the prototype chain when extending built-in classes (`Error` in this case). This is necessary for instance methods (like instanceof) to work correctly.